Controller/Callback/Offsite.php

```diff
@@ -110,19 +113,30 @@ public function execute()
             return $this->redirect(self::PATH_CART);
         }
 
-        if ($order->getState() === Order::STATE_PROCESSING) {
-            return $this->redirect(self::PATH_SUCCESS);
+        if (! $payment = $order->getPayment()) {
+            $this->invalid($order, __('Cannot retrieve a payment detail from the request. Please contact our
+            support if you have any questions.'));
+
+            return $this->redirect(self::PATH_CART);
         }
 
-        if ($order->getState() !== Order::STATE_PENDING_PAYMENT) {
-            $this->invalid($order, __('Invalid order status, cannot validate the payment. Please contact our
-            support if you have any questions.'));
+        $token = $this->request->getParam('token');
+
+        if (!$token || $payment->getAdditionalInformation('token') !== rtrim($token, "/")) {
+            $this->invalid(
+                $order,
+                __('Invalid token. Please contact our support if you have any questions.')
```

Comment:
I think this message is not too meaningful to the customer trying to pay. They may be confused by what token means. Can we say something like `Unknown error. Please contact our support if you have any questions`? also not as useful but won't make the customer confused what a token is used for

Reply:
or any other suggestions you might have 🙂

Reply:
Yeah you are right. `Invalid URL` might be better. I will ask the team for better suggestions. Thanks.